src/msg/msg_types.cc

```diff
@@ -84,9 +84,13 @@ bool entity_addr_t::parse(const char *s, const char **end, int default_type)
     start += 4;
     newtype = TYPE_ANY;
   } else if (*s == '-') {
+    const char* nextchar = s + 1;
+    if (*nextchar != ' ') {
+      return false;
```

Comment:
so we expect a `TYPE_NONE` address should end with a space. i don't quite understand how this is connected with assert failure in `entity_addr_t::set_port()` in https://tracker.ceph.com/issues/48336 . how does an NONE address kick in?

Reply:
@tchaikov because '--clobber' begins with a '-' and therefore it parses just fine as an entity_addr_t but fails miserably when you call set_port on it?